doc/cs_registers.rst

```diff
@@ -153,6 +181,30 @@ When an exception is encountered, this register can hold exception-specific info
 For all other exceptions, ``mtval`` is 0.
 
 
+Machine Interrupt Pending Register (mip)
+----------------------------------------
+
+CSR Address: ``0x344``
+
+Reset Value: ``0x0000_0000``
+
+``mip`` is a read-only register indicating pending interrupt requests.
+A particular bit in the register reads as one if the corresponding interrupt input signal is high and if the interrupt is enabled in the ``mie`` CSR.
+
++-------+---------------------------------------------------------------------------------------+
+| Bit#  | Interrupt                                                                             |
++-------+---------------------------------------------------------------------------------------+
+| 30:16 | Machine Fast Interrupts Pending: If bit x+16 is set,                                  |
+|       | fast interrupt ``irq_fast_i[x]`` is pending.                                          |
```

Comment:
The NMI logic needs to be documented here as well.

Reply:
According to the spec, the NMI is not visible in the mip CSR. Thus I did not include it here.